### PR TITLE
feat: tag assign region

### PR DESCRIPTION
**Pull request by wai-wong-edb**

### CODE REVIEW

File: pkg/provider/data_source_region.go [hunk 1]
```diff
@@ -90,6 +90,23 @@ func (r *regionsDataSource) Schema(ctx context.Context, req datasource.SchemaReq
 				Description: "Unique region ID. For example, \"germanywestcentral\" in the Azure cloud provider, \"eu-west-1\" in the AWS cloud provider.",
 				Optional:    true,
 			},
+			"tags": schema.SetNestedAttribute{
+				Description: "Assign existing tags or create tags to assign to this resource",
```
Comment (vishalsawale9) [inline pkg/provider/data_source_region.go:94]:
```suggestion
				Description: "Assign existing tags or create new tags to assign to this resource",
```

File: pkg/provider/resource_region.go [hunk 2]
```diff
@@ -83,6 +86,36 @@ func (r regionResource) Schema(ctx context.Context, req resource.SchemaRequest,
 					stringplanmodifier.UseStateForUnknown(),
 				},
 			},
+			"tags": schema.SetNestedAttribute{
+				Description: "Assign existing tags or create tags to assign to this resource",
```
Comment (vishalsawale9) [inline pkg/provider/resource_region.go:90]: Same as [above](https://github.com/EnterpriseDB/terraform-provider-biganimal/pull/582/files#r1728390020)

**Review by vishalsawale9 — COMMENTED** [review]
Left some minor comments. Rest looks good to me.

File: pkg/provider/data_source_region.go (hunk 1 above)
Comment (edbamolvyavahare) [inline pkg/provider/data_source_region.go:94]: as this is a data_source description we should not mention assign or create, instead it should mention something like "show tags associated with this resource"

Comment (wai-wong-edb) [reply pkg/provider/data_source_region.go:94]: I will change it all in a later PR https://github.com/EnterpriseDB/terraform-provider-biganimal/pull/583 so it is done altogether

**Review by edbamolvyavahare — APPROVED** [review]
LGTM

**Pull request merged**